PiwikTracker/MemoryQueue.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+
+/// The MemoryQueue is a **thread-unsave** in memory Queue.
+class MemoryQueue: Queue {
+    private var items = [Data]()
+    
+    var itemCount: Int { get {
+        return items.count
+        }
+    }
+    
+    func queue(item: NSCoding, completion: ()->()) {
+        let data = NSKeyedArchiver.archivedData(withRootObject: item)
```

Comment:
There is no need to serialise the items in this case, you can just store them as is if you make the class generic.

And the queue will return the same type as you put in.

````
/// The MemoryQueue is a **thread-unsave** in memory Queue.
struct MemoryQueue<E: NSCoding>: Queue {    
    typealias T = E    
    private var items: [E] = []
    ...    
}
````